cmake/Dependencies.cmake

```diff
@@ -1624,7 +1624,11 @@ if(NOT INTERN_BUILD_MOBILE)
 
   # use cub in a safe manner, see:
   # https://github.com/pytorch/pytorch/pull/55292
-  if(NOT ${CUDA_VERSION} LESS 11.5)
+  set(THRUST_SEARCH_PATHS ${CUDA_INCLUDE_DIRS})
+  list(TRANSFORM THRUST_SEARCH_PATHS APPEND /thrust/cmake)
```

Comment:
Looks like this directory does not exist?
```bash
ls /usr/local/cuda/include/thrust | grep cmake
```
gives no output on my installation.

Reply:
```
dpkg --contents cuda-thrust-11-3_11.3.109-1_amd64.deb  | grep cmake
drwxr-xr-x root/root         0 2021-05-04 04:21 ./usr/local/cuda-11.3/targets/x86_64-linux/include/cub/cmake/
-rw-r--r-- root/root      1526 2021-05-04 04:21 ./usr/local/cuda-11.3/targets/x86_64-linux/include/cub/cmake/cub-config-version.cmake
-rw-r--r-- root/root      2804 2021-05-04 04:21 ./usr/local/cuda-11.3/targets/x86_64-linux/include/cub/cmake/cub-config.cmake
drwxr-xr-x root/root         0 2021-05-04 04:21 ./usr/local/cuda-11.3/targets/x86_64-linux/include/thrust/cmake/
-rw-r--r-- root/root     16381 2021-05-04 04:21 ./usr/local/cuda-11.3/targets/x86_64-linux/include/thrust/cmake/FindTBB.cmake
-rw-r--r-- root/root     26883 2021-05-04 04:21 ./usr/local/cuda-11.3/targets/x86_64-linux/include/thrust/cmake/thrust-config.cmake
-rw-r--r-- root/root      1601 2021-05-04 04:21 ./usr/local/cuda-11.3/targets/x86_64-linux/include/thrust/cmake/thrust-config-version.cmake
-rw-r--r-- root/root      7818 2021-05-04 04:21 ./usr/local/cuda-11.3/targets/x86_64-linux/include/thrust/cmake/README.md
```
should targets/x86_64-linux not be in the CUDA_INCLUDE_DIRS?

Reply:
Look on CUDA 11.5, the path has changed:
```bash
[poincare] gaoxiang@sunnyvale ~/misc master $ pacman -Ql cuda | grep cmake
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/cub/
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/cub/cub-config-version.cmake
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/cub/cub-config.cmake
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/thrust/
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/thrust/FindTBB.cmake
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/thrust/README.md
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/thrust/thrust-config-version.cmake
cuda /opt/cuda/targets/x86_64-linux/lib/cmake/thrust/thrust-config.cmake
```
But I still prefer the compilation way because it seems to be more robust.

Reply:
Ohoh. Well, arguably, it's an improvement over putting it in include... 
So the snippet for the compilation seems very elaborate and doesn't work for cross-compilation, while adding include + lib search paths seems to be more simple.

Reply:
OK, I didn't think about cross-compilation. Then you can keep the find_package way.

Reply:
Thank you!
Hopefully, the updated search path finds thrust on 11.5, too.